### PR TITLE
feat: barista perks

### LINKED ISSUE #422: Find a permanent solution to getting barista products
As per the barista hub change, the barista products are hardcoded in the app in `/lib/data/repositories/barista_product/barista_product_repository.dart`

These should be fetched from the backend, after we have marked barista products in the database

### PR DESCRIPTION
**Pull request by fredpetersen**
Closes #422 

Visual: A new section has been added to the tickets page, that shows your perks if your user is registered as Barista+

All purchaseable products are now loaded in the SplashRouter which has now been renamed to MainRedirectionRouter (These products are now also fetched from ProductsControllerV2 instead of V1)

### CODE REVIEW

File: lib/cubits/user/user_cubit.dart [hunk 1]
```diff
@@ -16,6 +16,14 @@ class UserCubit extends Cubit<UserState> {
   UserCubit(this._accountRepository, this._occupationRepository)
       : super(UserLoading());
 
+  Future<void> init() async {
+    await refreshUserDetails();
+    final st = state;
+    if (st is UserLoaded) {
+      emit(UserInitiallyLoaded(st));
```
Comment (fremartini) [inline lib/cubits/user/user_cubit.dart:23]: if (state is UserLoaded) { emit(UserInitiallyLoaded(state)); } ?

File: lib/models/ticket/ticket_count.dart [hunk 2]
```diff
@@ -1,3 +1,5 @@
+const _baristaProductIds = [9, 11, 12, 13];
```
Comment (fremartini) [inline lib/models/ticket/ticket_count.dart:1]: ??????

File: lib/utils/user_group_permissions.dart [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import 'package:coffeecard/models/account/user.dart';
+
+bool hasBaristaPerks(User user) {
+  switch (user.userGroup) {
```
Comment (fremartini) [inline lib/utils/user_group_permissions.dart:4]: return user.userGroup in [UserGroup.Barista, ..., ... ,], default false

File: lib/widgets/components/tickets/tickets_section.dart [hunk 4]
```diff
@@ -15,9 +16,27 @@ import 'package:coffeecard/widgets/components/tickets/coffee_card_placeholder.da
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
+class BaristaTicketSection extends StatelessWidget {
+  const BaristaTicketSection();
+
+  @override
+  Widget build(BuildContext context) {
+    return _TicketSection(filter: (ticket) => ticket.isBaristaTicket());
```
Comment (fremartini) [inline lib/widgets/components/tickets/tickets_section.dart:24]: why underscore?

File: lib/core/widgets/components/barista_perks_section.dart [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+          children: [
+            SectionTitle(Strings.baristaPerks),
+            BaristaIndicator(),
+          ],
+        ),
+        Grid(
+          singleColumnOnSmallDevice: false,
+          gap: GridGap.tightVertical,
+          gapSmall: GridGap.tight,
+          children: [
```
Comment (marfavi) [inline lib/core/widgets/components/barista_perks_section.dart:35]: List should not be hardcoded but rather based on perks available to the user

File: lib/core/widgets/pages/home_page.dart [hunk 6]
```diff
@@ -41,6 +45,33 @@ class _HomePageState extends State<HomePage> {
       ..add(newFlowIndex);
   }
 
+  List<AppFlow> get _bottomNavAppFlows => <AppFlow>[
```
Comment (marfavi) [inline lib/core/widgets/pages/home_page.dart:48]: Why was this moved up the file?

File: lib/data/repositories/barista_product/barista_product_repository.dart [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import 'package:coffeecard/features/environment/domain/entities/environment.dart';
+import 'package:coffeecard/features/environment/presentation/cubit/environment_cubit.dart';
+import 'package:coffeecard/service_locator.dart';
+
+class BaristaProductsRepository {
+  List<int> getBaristaProductIds() {
+    // TODO(fredpetersen): Replace with data fetch from backend, https://github.com/AnalogIO/coffeecard_app/issues/422
```
Comment (marfavi) [inline lib/data/repositories/barista_product/barista_product_repository.dart:7]: Work on this TODO

File: lib/features/product/data/models/product_model.dart [hunk 8]
```diff
@@ -17,6 +18,7 @@ class ProductModel extends Product {
       name: dto.name,
       id: dto.id,
       description: dto.description,
+      isPerk: false, // TODO Integrate with backend
```
Comment (marfavi) [inline lib/features/product/data/models/product_model.dart:21]: Work on this TODO

**Review by marfavi — CHANGES_REQUESTED** [review]
Half-review so we know what to work on next

**Comment by marfavi** [comment]
I'd suggest to move the "loading the user in the splash screen " functionality into its own PR to speed things up

File: ios/Podfile [hunk 9]
```diff
@@ -40,6 +40,10 @@ post_install do |installer|
       config.build_settings['CODE_SIGNING_REQUIRED'] = "NO"
       config.build_settings['CODE_SIGNING_ALLOWED'] = "NO"
       config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'
+      xcconfig_path = config.base_configuration_reference.real_path
```
[the remark this answers is not in the document]
Comment (marfavi) [reply ios/Podfile]: Done

File: lib/base/style/text_styles.dart [hunk 10]
```diff
@@ -0,0 +1 @@
+
```
[the remark this answers is not in the document]
Comment (marfavi) [reply lib/base/style/text_styles.dart]: Done

**Comment by marfavi** [comment]
> I'd suggest to move the "loading the user in the splash screen " functionality into its own PR to speed things up

Done via #528

File: lib/core/strings.dart [hunk 11]
```diff
@@ -139,7 +147,6 @@ abstract final class Strings {
       'Tickets that you buy will show up here.';
   static const emptyCoffeeCardTextBottom =
       'Use the section below to shop tickets.';
-  static const cofeeCardTicketsLeft = 'Tickets left:';
```
[the remark this answers is not in the document]
Comment (fredpetersen) [reply lib/core/strings.dart:142]: It is unused, i dont want to create a seperate PR for this one line :)

File: lib/core/widgets/components/barista_indicator.dart [hunk 12]
```diff
@@ -2,18 +2,21 @@ import 'package:coffeecard/core/strings.dart';
...
+
+  final Role userRole;
 
   @override
   Widget build(BuildContext context) {
     return TextButton(
       onPressed: () => appDialog(
         context: context,
-        title: 'Barista',
+        title: userRole.name, // TODO(fredpetersen): Should be capitalized
```
Comment (marfavi) [inline lib/core/widgets/components/barista_indicator.dart:19]: use `.capitalize()` extension method to do this

File: lib/core/widgets/components/barista_indicator.dart [hunk 13]
```diff
@@ -45,7 +48,10 @@ class BaristaIndicator extends StatelessWidget {
       child: Row(
         mainAxisSize: MainAxisSize.min,
         children: [
-          Text('Barista', style: AppTextStyle.baristaButton),
+          Text(
+            userRole.name, // TODO(fredpetersen): Should be capitalized
```
Comment (marfavi) [inline lib/core/widgets/components/barista_indicator.dart:52]: use .capitalize() extension method to do this

**Review by marfavi — COMMENTED** [review]
Looks good

**Review by marfavi — APPROVED** [review]

**Review by marfavi — APPROVED** [review]

**Pull request merged**